lavalink/node.py

```diff
@@ -176,39 +176,47 @@ async def _multi_try_connect(self, uri):
         backoff = ExponentialBackoff()
         attempt = 1
 
-        while self._is_shutdown is False and (self._ws is None or not self._ws.open):
+        while self._is_shutdown is False and (self._ws is None or self._ws.closed):
             try:
-                ws = self._ws = await websockets.connect(uri, extra_headers=self.headers)
-                return ws
+                ws = await self.session.ws_connect(url=uri, headers=self.headers)
             except OSError:
                 delay = backoff.delay()
                 log.debug("Failed connect attempt %s, retrying in %s", attempt, delay)
                 await asyncio.sleep(delay)
                 attempt += 1
```

Comment:
From what I checked, `aiohttp.WSServerHandshakeError` should be raised in situations where `websockets.InvalidStatusCode` was raised before
```suggestion
                attempt += 1
            except aiohttp.WSServerHandshakeError:
                return None
```